### DISCUSSION TITLE
How to save regex history

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#2451] [Q&A] [answered]
I've been working with VisiData on a new machine. For some reason when I do a regex search, for example a column search, VisiData doesn't retain the history of the searches beyond the current session. This means that when I quit VisiData and then start it again, and I do a regex search and press up to see the last search, it doesn't show anything.

I remember this happened about a year ago and there may have been a solution, but I don't remember what it is. Do you?

**Comment by midichef** [comment, marked as answer]
Try creating the `.visidata` directory:  `mkdir ~/.visidata`.
(more specifically, whatever directory is specified by the `visidata_dir` option in the global options sheet opened with `Shift+O`)
This worked for me on a fresh install of visidata v3.1dev.

**Reply by cool-RR** [reply]
Works! Thank you.
